tests/i18n/patterns/tests.py

```diff
@@ -92,6 +93,18 @@ def test_prefixed_i18n_disabled(self):
             self.assertEqual(reverse('prefixed'), '/prefixed/')
 
 
+class RequestURLConfTests(URLTestCaseBase):
```

Comment:
I wonder if we need `URLTestCaseBase` -- I think `@override_settings(ROOT_URLCONF=` should take care of that cache clearing.

It seems like the existing test organization isn't very good. It would be nice to organize all the tests for `LocaleMiddleware` together (obviously not the job of this PR). I just wonder if we'll have other tests added to this new test class in the future.